File: python/chapter_1/is_unique.py
def is_unique(s):
    """Return whether or not string contains all unique characters.

    >>> is_unique("uniqe ya")
    True
    >>> is_unique("unique ya")
    False
    """
    seen = {}
    for c in s:
        if c in seen: return False
        seen[c] = True
    return True


if __name__ == "__main__":
    import doctest
    doctest.testmod(verbose=True)

File: ruby/ctci/chapter_1/is_unique.rb
# @example
#   is_unique("uniqe ya") #=> true
#   is_unique("unique ya") #=> false
def is_unique(str)
  seen = Hash.new
  str.each_char { |c|
    if seen.member?(c)
      return false
    else
      seen[c] = true
    end
  }
  return true
end
